### PR TITLE
few mypy fixes

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by edgarriba** [comment]
@shijianjian review and feel free to take it. I'd suggest to invest some time now the Union works for jit, that we could remove all the type:ignore in the augmentations and have at least good tsting on that side too. 

Besides, recently applying better naming where we have optional:

```python
foo = Tensor
maybe_foo = Optional[Tensor] = None
# or 
maybe_foo = Tensor([0])
```

then we could have something like
```python
foo: Tensor = KORNIA_UNWRAP(maybe_foo)
```
that takes care already to cast in the correct type.

Learning this tricks form c++ side and make everything more readable and type safe.

File: kornia/augmentation/container/augment.py [hunk 1]
```diff
@@ -242,13 +237,18 @@ def inverse(  # type: ignore
                 input = arg.data  # all boxes are in (B, N, 4, 2) format now.
             else:
                 input = arg
-            for (name, module), param in zip_longest(list(self.get_forward_sequence(params))[::-1], params[::-1]):
+            for (_, module), param in zip_longest(list(self.get_forward_sequence(params))[::-1], params[::-1]):
                 if isinstance(module, (_AugmentationBase, ImageSequential)):
-                    param = params[name] if name in params else param
+                    # TODO(jian): verify what's happening here. Mypy was complaining,
+                    # param = params[name] if name in params else param
```
Comment (edgarriba) [inline kornia/augmentation/container/augment.py:243]: @shijianjian please check this — seems to pass test but not confident what’s happening

Comment (shijianjian) [reply kornia/augmentation/container/augment.py:243]: This shouldn't pass. It consumes wrong param but still reproducible I guess.

Comment (edgarriba) [reply kornia/augmentation/container/augment.py:243]: Params is typed as list but in practice seems to be a dict

File: kornia/augmentation/container/augment.py [hunk 2]
```diff
@@ -242,13 +237,18 @@ def inverse(  # type: ignore
...
+                    pass
```
Comment (shijianjian) [inline kornia/augmentation/container/augment.py:244]:
```suggestion
                if isinstance(module, (_AugmentationBase,)):
                    param = params[name]
                elif isinstance(module, (ImageSequential,)):
                    pass
```

File: kornia/augmentation/container/augment.py [hunk 3]
```diff
@@ -242,13 +237,18 @@ def inverse(  # type: ignore
                 input = arg.data  # all boxes are in (B, N, 4, 2) format now.
             else:
                 input = arg
-            for (name, module), param in zip_longest(list(self.get_forward_sequence(params))[::-1], params[::-1]):
+            for (_, module), param in zip_longest(list(self.get_forward_sequence(params))[::-1], params[::-1]):
                 if isinstance(module, (_AugmentationBase, ImageSequential)):
```
Comment (shijianjian) [inline kornia/augmentation/container/augment.py:241]:
```suggestion
```

Comment (shijianjian) [inline kornia/augmentation/container/augment.py:241]:
```suggestion
```

**Review by shijianjian — COMMENTED** [review]
The original code is okay I think. Probably something like this will mypy happier.

File: kornia/augmentation/container/augment.py (hunk 2 above)
Comment (edgarriba) [reply kornia/augmentation/container/augment.py:244]: i think the point is that why by commenting that small piece of code tests are still passing

**Pull request merged**